homeassistant/components/smarttub/sensor.py

```diff
@@ -2,17 +2,49 @@
 from enum import Enum
 import logging
 
+import smarttub
+import voluptuous as vol
+
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import config_validation as cv, entity_platform
+
 from .const import DOMAIN, SMARTTUB_CONTROLLER
 from .entity import SmartTubSensorBase
 
 _LOGGER = logging.getLogger(__name__)
 
+
 ATTR_DURATION = "duration"
 ATTR_LAST_UPDATED = "last_updated"
 ATTR_MODE = "mode"
 ATTR_START_HOUR = "start_hour"
 
 
+SET_PRIMARY_FILTRATION_SCHEMA = vol.Schema(
+    vol.All(
+        cv.has_at_least_one_key(ATTR_DURATION, ATTR_START_HOUR),
+        {
+            vol.Required(ATTR_ENTITY_ID): cv.entity_id,
```

Comment:
This is handled by the `async_register_entity_service` and not needed in this schema.

```suggestion
```

The entity service makes sure the right entities are called.